controllers/actions.summerwind.net/runner_controller.go

```diff
@@ -1001,6 +1002,35 @@ func newRunnerPodWithContainerMode(containerMode string, template corev1.Pod, ru
 			)
 		}
 
+		// explicitly invoke `dockerd` to avoid automatic TLS / TCP binding
+		dockerdContainer.Args = append([]string{
+			"dockerd",
+			"--host=unix:///run/docker/docker.sock",
+		}, dockerdContainer.Args...)
+
+		// this must match a GID for the user in the runner image
+		// default matches GitHub Actions infra (and default runner images
+		// for actions-runner-controller) so typically should not need to be
+		// overridden
+		if ok, _ := envVarPresent("DOCKER_GROUP_GID", dockerdContainer.Env); !ok {
+			dockerdContainer.Env = append(dockerdContainer.Env,
+				corev1.EnvVar{
+					Name: "DOCKER_GROUP_GID",
+					Value: "121",
```

Comment:
If we go this way, we might need to make this configurable because folks can use another gid for the docker user if they use a custom runner image(and we don't forbid the use of a custom runner image...

Reply:
I just realized you already addressed that by checking for the existence of DOCKER_GROUP_GID and only use the hard-coded 121 when the envvar does not exist!
Good work 👍

Reply:
Yeah, this is unfortunately still not the most elegant, but I couldn't figure out a cleaner way to handle this, as we do need to pass a GID so that the socket is usable.